### PR TITLE
tools, github: Pick authorization prefix based on provided token

### PR DESCRIPTION
**Pull request by dio**
Commit Message: Pick the authorization prefix based on the provided token. 
Additional Description: Reference: https://github.com/octokit/auth-token.js/blob/902a172693d08de998250bf4d8acb1fdb22377a4/src/with-authorization-prefix.ts#L6-L12
Risk Level: Low
Testing: Manual
Docs Changes: N/A
Release Notes: N/A
Platform-Specific Features: N/A

Signed-off-by: Dhi Aurrahman <dio@rockybars.com>

### CODE REVIEW

**Comment by dio** [comment]
Hopefully, this works now.

File: tools/github/write_current_source_version.py [hunk 1]
```diff
@@ -68,10 +68,12 @@
     # Fetch the current version commit information from GitHub.
     commit_info_request = urllib.request.Request(
         "https://api.github.com/repos/envoyproxy/envoy/commits/v" + current_version)
-    github_token = os.environ.get(args.github_api_token_env_name)
-    if github_token != None:
+    if github_token := os.environ.get(args.github_api_token_env_name):
+        # Reference: https://github.com/octokit/auth-token.js/blob/902a172693d08de998250bf4d8acb1fdb22377a4/src/with-authorization-prefix.ts#L6-L12.
+        authorization_header_prefix = "bearer" if len(github_token.split(".")) is 3 else "token"
```
Comment (phlax) [inline tools/github/write_current_source_version.py:73]:
```suggestion
        authorization_header_prefix = "bearer" if len(github_token.split(".")) == 3 else "token"
```

```console
F632 use ==/!= to compare constant literals (str, bytes, int, float, tuple)
```

i think our flake8 is a bit limited and needs to have more rules added

**Review by phlax — APPROVED** [review]
lgtm, thanks @dio 

im wondering if we need to add some info to the README about using a token to generate the `SOURCE_VERSION` file

also wondering about error handling - its very common for tokens to expire for example

**Comment by dio** [comment]
Sure. Will add. When you do it locally seems like it is rare to experience rate-limited API calls for commit info (e.g. https://api.github.com/repos/envoyproxy/envoy/commits/v1.24.0). But yeah when you do from CI, e.g. GitHub Actions it has a bigger probability. Also, +1 for clean error handling.

**Comment by phlax** [comment]
cool - i can review any follow ups - ill wait on the backport pr/s until you are happy with this on `main`

**Pull request merged**